docs/_basic/ja_authenticating_oauth.md

```diff
@@ -6,16 +6,125 @@ order: 15
+Slack アプリの配布準備のために、 Bolt による OAuth フローを実装し、インストール時に取得した情報をセキュアな方法で保存しておく必要があります。
+Bolt は OAuth による認証そのものと、OAuth ルーティング、 state にる認証、保存すべきインストール情報の受け渡しなどの周辺機能もサポートしています。
 
+OAuth を有効にするために必要なもの:
+* `clientId`, `clientSecret`, `stateSecret`, `scopes` _(必須)_
+* `installationStore` は、データベースへのアプリ情報の読み込みと書き込みをするハンドラーのオプションです *(任意、プロダクションコードにおいては強く推奨)*
 
-Bolt for JavaScript は、アプリのインストールフローを完了した後の遷移先の URL である **Redirect URL** のためのパスとして `slack/oauth_redirect` を有効にします。この URL を Slack アプリの設定画面の **OAuth and Permissions** セクション内で **Redirect URL** として指定してください。この設定のカスタマイズは以下で説明する `installerOptions` を指定することで可能です。
+##### 開発とテスト
 
-Bolt for JavaScript は `slack/install` というパスも生成します。これは、Slack アプリのダイレクトインストールのために `Add to Slack` ボタンを置く場合に指定できる URL です。アプリはすでにインストールされていて、さらにユーザーから追加の認可情報（例：ユーザートークンの発行）な場合や、何らかの理由で動的にインストール用の URL を生成したい場合は、`ExpressReceiver` を自前でインスタンス化し、それを `receiver` という変数に代入した上で `receiver.installer.generateInstallUrl()` を呼び出してください。詳細は [OAuth ライブラリのドキュメント](https://slack.dev/node-slack-sdk/oauth#generating-an-installation-url)の `generateInstallUrl()` を参照してください。
+開発とテストにおいて利用可能な `installationStore` オプションの実装 `FileInstallationStore` を Bolt は既に用意しています。
 
-Slack の OAuth インストールフローについてもっと知りたい場合は [API ドキュメント](https://api.slack.com/authentication/oauth-v2)を参照してください。
+```javascript
+const { App } = require('@slack/bolt');
+const { FileInstallationStore } = require('@slack/oauth');
+const app = new App({
+  signingSecret: process.env.SLACK_SIGNING_SECRET,
+  clientId: process.env.SLACK_CLIENT_ID,
+  clientSecret: process.env.SLACK_CLIENT_SECRET,
+  stateSecret: 'my-state-secret',
+  scopes: ['channels:history', 'chat:write', 'commands'],
+  installationStore: new FileInstallationStore(),
+});
+```
+:warning: プロダクションコードとしての利用は **推奨しません** ので、自前で実装する必要があります。サンプルコードとして [OAuth の別の実装例](https://github.com/slackapi/bolt-js/tree/main/examples/oauth)を参照してください。
+
+##### アプリのインストール
+
+* **インストール時の初期化**: Bolt for JavaScript は `slack/install` という **パス** を生成します。これは、Slack アプリのダイレクトインストールのために `Add to Slack` ボタンを置く場合に指定できる URL です（ `state` パラメータが必要）。例えば、 _www.example.com_ でホスティングされているアプリであれば、 _www.example.com/slack/install_ とインストールページが提供されます。
+  * 💡 `App` コンストラクタ内で `installerOptions.directInstall: true` と設定することで、デフォルトのウェブページを描画しない代わりに、ユーザーへ Slack の認証 URL を出すことができます（[例](https://github.com/slackapi/bolt-js/blob/5b4d9ceb65e6bf5cf29dfa58268ea248e5466bfb/examples/oauth/app.js#L58-L64)）。
+
+* **Slack へ追加**: `Add to Slack` ボタンを押すと Slack との OAuth プロセスを開始します。ユーザーがアプリへの権限付与を許可した後、 Slack はアプリの **Redirect URI** （既に設定済み）を呼び出し, ユーザーに **Open Slack** を促します。詳細に設定を変えたい場合は **Redirect URI** セクションを参照してください。
```

Comment:
"Slack will call your app’s Redirect URI" can sound a bit unclear as it's actually redirection. I will update the English one later on.

```suggestion
* **Add to Slack (Slack へ追加)**: `Add to Slack` ボタンを押すと Slack との OAuth プロセスを開始します。ユーザーがアプリへの権限付与を許可すると、Slack はアプリの **Redirect URI** （あらかじめ設定されています）へユーザーを誘導し、処理が正常に完了したらユーザーに **Slack で開く** よう促します。これらの設定をカスタマイズする方法については、後述の **Redirect URI** セクションを参照してください。
```